### ISSUE TITLE
The field named `js` in the schema causes the Relay compiler to fail when this field is requested in the query

### ISSUE THREAD

**Issue by SiarheiBarysenka** [facebook/relay#4722]
## Problem

The field named `js` in the schema causes the Relay compiler to fail when this field is requested in the query.

## Steps to reproduce

1. Define the following schema:

```
type Query {
  hello: String
  js: String
}

schema {
  query: Query
}
```

2. Define the following query:

```
const query = graphql`
  query srcAppQuery {
    hello
    js
  }
`;
```

3. Execute the Relay compiler to generate the query and observe the output in the console.

## Expected Result

The query is generated successfully. Expected output in the console:

```
> bug-relay-compiler-js-field@1.0.0 relay
> relay-compiler

[INFO] [default] compiling...
[INFO] [default] compiled documents: 1 reader, 1 normalization, 1 operation text
[INFO] Compilation completed.
[INFO] Done.
```

## Actual Result

The query generation fails. Actual output in the console:

```
> bug-relay-compiler-js-field@1.0.0 relay
> relay-compiler

[INFO] [default] compiling...
[ERROR] Error: ✖︎ 'js' should be defined on the server schema.

  src/index.ts:6:5
    5 │     hello
    6 │     js
      │     ^^
    7 │   }

[ERROR] Compilation failed.
[ERROR] Unable to run relay compiler. Error details: 
Failed to build:
 - Validation errors: 1 error(s) encountered above.
```

## Notes

1. If the `js` field is removed from the definition of `srcAppQuery`, the query is generated successfully (but obviously without the `js` field included).
2. The sample project to reproduce the issue is available at https://github.com/SiarheiBarysenka/bug-relay-compiler-js-field .

**Comment by captbaritone** [comment]
I was able to [reproduce in the compiler explorer](https://relay.dev/compiler-explorer/#enc=1&schemaText=C4TwDgpgBAigrhATiKBvAUFKALCAbPAewC4oBlYRASwDsBzTKAKwGdSLr70BfddFgMa4AtgEM0jAI4JkpeEhA8gA&documentText=I4VwpgTgngBAzhAxgQQA6oIrmjA3gKBhgAswAbMge0JgCs58BfIA&inputWindow=document&outputType=operation&no_inline=true&enable_3d_branch_arg_generation=true&actor_change_support=true&text_artifacts=true&skip_printing_nulls=false&compact_query_text=false&enable_fragment_aliases=false&enforce_fragment_alias_where_ambiguous=false&enable_catch_directive_transform=false&disallow_required_on_non_null_fields=false&language=typescript).

Likely some bug in our `@match` validation [here](https://github.com/facebook/relay/blob/fc5894ab7aa298340babd65d8e0d85fce8e0bee8/compiler/crates/relay-transforms/src/match_/match_transform.rs#L176-L193)

**Comment by SiarheiBarysenka** [comment]
Thank you @captbaritone 🙏 Is this something that can be addressed in the foreseeable future?

**Comment by SiarheiBarysenka** [comment]
Hey @captbaritone I wonder if there's any update on this? Thanks!

**Comment by SiarheiBarysenka** [comment]
Hey @captbaritone Is there any updates on the fix timelines for this issue? Thanks.

**Comment by captbaritone** [comment]
I don't think we have active plans to work on this this, but folks are welcome to look at the code pointers I shared and work on a fix or at least narrowing down where the issue is. Sorry we don't have bandwidth to fix right now.
